src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,6 +150,115 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
             return files;
         }
 
+        /// <summary>
+        /// Gets auth plugins installed using dotnet tools. This is done by iterating through each file in directories found int eh
+        /// `PATH` environment variable.
+        /// The files are also expected to have a name that starts with `nuget-plugin-`
+        /// </summary>
+        /// <returns></returns>
+        private List<PluginFile> GetNetToolsPluginFiles()
+        {
+            var pluginFiles = new List<PluginFile>();
+
+            if (_environmentVariableReader is null)
+            {
+                _environmentVariableReader = EnvironmentVariableWrapper.Instance;
+            }
+
+            var paths = _environmentVariableReader.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) ?? Array.Empty<string>();
```

Comment:
In the [spec](https://github.com/NuGet/Home/blob/dev/accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md?plain=1#L165-L172), we have mentioned that, if customers prefer to install NuGet plugins as a [tool-path global tool](https://learn.microsoft.com/dotnet/core/tools/global-tools-how-to-use#use-the-tool-as-a-global-tool-installed-in-a-custom-location), they can set the `NUGET_PLUGIN_PATHS` environment variable.
This variable should point to the location of the .NET Tool executable.

I think we are not handling this case in this PR. Please let me know if I am missing something.

Reply:
Good catch